src/extensions/FileUploadExt.ts

```
@@ -0,0 +1,2205 @@
+                        reject(error);
+                    }
+                };
+                
+                reader.onerror = () => reject(new Error('Error reading image file'));
+                reader.readAsDataURL(file);
+            } catch (error) {
+                reject(error);
+            }
+        });
+    },
+    
```

Biome (1.9.4)

[error] Promise executor functions should not be `async`.



(lint/suspicious/noAsyncPromiseExecutor)